web/ui/react-app/src/pages/Services.tsx

```diff
@@ -1,15 +1,199 @@
+    }
+
+    return labels;
+  };
+
+  render() {
+    return (
+      <>
+        <h2>Service Discovery</h2>
+        <ul>
+          {Object.keys(this.state.targets).map((val, i) => (
+            <li key={Object.keys(this.state.targets)[i]}>
+              <a href={'#' + val}>
+                {' '}
+                {val} ({this.state.targets[val].active} / {this.state.targets[val].total} active targets){' '}
+              </a>
+            </li>
+          ))}
+        </ul>
+        <hr />
+        <div className="outer-layer">
+          {Object.keys(this.state.labels).map((val: any, i) => {
+            const value = this.state.labels[val];
+            return (
+              <div id={val} key={Object.keys(this.state.labels)[i]} className="label-component">
+                <span className="target-head">
+                  {' '}
+                  {i + 1}. {val}{' '}
+                </span>
+                <Labels value={value} />
+              </div>
+            );
+          })}
+        </div>
+      </>
+    );
+  }
+}
+
+class Labels extends Component<LabelProps, LabelState> {
+  constructor(props: LabelProps) {
+    super(props);
+
+    this.state = {
+      showMore: false,
+    };
+  }
+
+  toggleMore = () => {
+    this.setState({ showMore: !this.state.showMore });
+  };
+
+  render() {
+    return (
+      <>
+        {!this.state.showMore ? (
+          <button className="btn btn-primary btn-sm" onClick={this.toggleMore}>
+            More
+          </button>
+        ) : (
+          <>
+            <button className="btn btn-primary btn-sm" onClick={this.toggleMore}>
+              Less
+            </button>
+            <div>
+              {Object.keys(this.props.value).map((_, i) => {
+                return (
+                  <div key={'inner-layer-' + i} className="row inner-layer">
+                    <div className="col-md-6">
+                      {i === 0 ? <div className="head">Discovered Labels</div> : null}
+                      <div key={this.props.value[i].discoveredLabels[i] + '-' + i}>
+                        {Object.keys(this.props.value[i].discoveredLabels).map((v, _) => (
+                          <div className="label-style" key={this.props.value[i].discoveredLabels[v]}>
+                            {' '}
+                            {v}={this.props.value[i].discoveredLabels[v]}{' '}
+                          </div>
+                        ))}
+                      </div>
+                    </div>
+                    <div className="col-md-6">
+                      {i === 0 ? <div className="head">Target Labels</div> : null}
+                      <div key={this.props.value[i].labels[i] + '-' + i}>
+                        {Object.keys(this.props.value[i].labels).map((v, _) => (
+                          <div className="label-style" key={this.props.value[i].labels[v]}>
+                            {' '}
+                            {v}={this.props.value[i].labels[v]}{' '}
```

Comment:
Can you please use Badge component here as it is in https://github.com/prometheus/prometheus/blob/master/web/ui/react-app/src/pages/targets/TargetLabels.tsx